addons/website/controllers/main.py

```diff
@@ -98,22 +98,21 @@ def robots(self):
 
     @http.route('/sitemap.xml', type='http', auth="public", website=True)
     def sitemap_xml_index(self):
-        cr, uid, context = request.cr, openerp.SUPERUSER_ID, request.context
-        ira = request.registry['ir.attachment']
-        iuv = request.registry['ir.ui.view']
-        mimetype ='application/xml;charset=utf-8'
+        Ira = request.env['ir.attachment'].sudo()
```

Comment:
hello @rim-odoo , Here You can see on line : https://github.com/odoo/odoo/pull/8230/files#diff-167e2f4c21937f0a924ebe41ed38b16dL101 simply uid is SUPERUSER_ID so used sudo.